File: restApiGo/controller/internal.go
package controller

import (
    "github.com/gin-gonic/gin"
    "net/http"
)

func RegisterInternalRoutes(r *gin.Engine) {
    r.GET("/api/internal/process", func(c *gin.Context) {
        var sum uint64 = 0
        for i := 0; i < 500_000_000; i++ {
            sum += uint64(i)
        }
        c.JSON(http.StatusOK, gin.H{"result": sum})
    })
}

File: restApiRust/src/controllers/internal.rs
use actix_web::{get, HttpResponse};

#[get("/api/internal/process")]
pub async fn process() -> HttpResponse {
    let mut sum: u64 = 0;
    for i in 0..500_000_000 {
        sum += i;
    }
    HttpResponse::Ok().json(serde_json::json!({ "result": sum }))
}
